### DISCUSSION TITLE
Attribute [stripeWebhooks] does not exist.

### DISCUSSION THREAD

**Discussion by mauriziofonte** [spatie/laravel-stripe-webhooks#124] [General]
Hello,

I'm following the installation guide as per documentation.

I'm arrived here: _Finally, take care of the routing: At the Stripe dashboard you must configure at what url Stripe webhooks should hit your app. In the routes file of your app you must pass that route to Route::stripeWebhooks:_

I've so added the provided code to my BaseController that handles Route creation.

```
<?php
namespace App\Http\Realms\Base;

use Illuminate\Foundation\Support\Providers\EventServiceProvider as AppProvider;
use Illuminate\Support\Facades\Route;
use Illuminate\Http\Request;

class BaseProvider extends AppProvider
{
    public function register()
    {
        // define all Base routes
        Route::group([
            'namespace'  => 'App\Http\Realms\Base',
            'as'         => 'main.',
            'middleware' => 'web',
        ], function () {

            // Homepage
            Route::get(
                '/',
                ['uses' => 'BaseController@home', 'as' => 'home']
            );
             .....
        });

        // redirect to 410 the default Cachier Stripe Webhooks
        Route::post('webhooks/stripe', function () {
            return abort(410);
        });

        // handle the correct Stripe Webhooks with Spatie/Laravel-Stripe-Webhooks
        Route::stripeWebhooks('handle/stripe/webhooks');

        // handle all other URLS in fallback mode
        Route::fallback('App\Http\Realms\Base\BaseController@fallback');
    }
}
```

Unfortunately, the app starts to respond with `Attribute [stripeWebhooks] does not exist.` for every request, php artisan included.

In my IDE (VSCode) I can see that a method named **stripeWebhooks** exists in `Illuminate\Support\Facades\Route::stripeWebhooks`

Is this an issue or a misconfiguration of some sort?

**Comment by phamamdzhiev** [comment]
Any luck with this issue? I am facing the same problem.

**Reply by mauriziofonte** [reply]
Unfortunately nope, I've reverted to the standard Cashier webhook handler and implemented my custom Event Listener around it.
At the end I've used portions of this package only to create the WebHook model to take care of non-unique webhook calls, but removed the logic to store the payload into DB because this is causing problems on MariaDB crashing on restart (and in my opinion it's generally not a good idea to store this load of data if the website it's going to have thousands of webhook calls per hour -- it should be an opt-in logic via config, not a pre-defined one)

**Comment by robert-watkin** [comment]
I'm also having this issue. All was fine in my local environment. Now I have the site on my dev server for testing I am getting the 'Attribute [stripeWebhooks] does not exist]'. I can't for the life of me figure out how to fix it

**Reply by mauriziofonte** [reply]
Hello Robert, unfortunately as I've commented on 2022-06-21 - [see above](https://github.com/spatie/laravel-stripe-webhooks/discussions/124#discussioncomment-2993328) - I've reverted in implementing my own Event Listener to handle Stripe Webhooks. 4 days of work was far better than being stuck in a non-resolving issue and losing much more in the long run. Also, Spatie does not seem to be responsive on this type of issues. Spatie is a great package creator and mantainer, but I think we're left alone if problems arise.

**Comment by hondaman900** [comment]
@RobertWatkin  Did you resolve this issue? I pushed my site live this morning and **the whole site broke**, same issue as you're having. Everything works great in local dev mode using Stripe CLI, but pushing to my Forge server I get a can't-deploy error and this in the logs:

```
In RouteRegistrar.php line 92:
  Attribute [stripeWebhooks] does not exist.
```

Did you resolve this or find the issue? I don't have the option of changing everything to Cashier.

Thanks in advance.

**Reply by mauriziofonte** [reply]
Hello [Stephen](https://github.com/hondaman900), no-one from Spatie ever looked into this issue and tried to give feedback. 

I'm sorry the whole website went down in prod, but you have no other option than try to replicate the issue in a _pseudo-prod_ environment and try debugging by tweaking around vendor files and dumping data at strategic points. I've spent days in trying to find the entry point of this problem, with no luck. 

At the end, I've re-implemented it. At least, I have control.

**Reply by hondaman900** [reply]
Thank you @mauriziofonte. In my case it looks like Composer ran out of memory trying to install `laravel-stripe-webhooks` and crashed, leaving the site broken with that error. I'm working with Forge tech support to try to resolve this, hopefully soon. I'll report back if this was the core issue or a coincidental event.

**Reply by hondaman900** [reply]
FWIW, turns out that for me it in fact was Composer running out of memory when installing a new package. Creating a new `composer.lock` on my local dev (using `composer update`) and then committing that to the live server (after much tweaking to resolve Git sync issues) fixed the issue.

**Reply by hondaman900** [reply]
Unfortunately now I'm stuck with an invalid signature error in production that's preventing my Stripe payment processing from working and I can't find the reason for this. It works in local dev, but on my live server it fails the signature test. Keys are present and valid. I guess I'll keep digging until I unearth a solution...

**Reply by anton-shynkarenko** [reply]
@hondaman900 did Webhook details page"Signing secret" Reveal action help you to solve the issue?
